AttendanceSystem.Domain.Services/AttendanceService.cs

```diff
@@ -40,6 +40,13 @@ public async Task<List<Session>> GetUpcomingSessionsForUser(string userId)
             .Where(s => s.Course.Students.Contains(userId)).ToListAsync();
     }
 
+    public async Task<List<Session>> GetUpcomingSessionsForTeacher(string userId)
+    {
+        return await _context.Sessions
+            .Include(s => s.Course)
+            .Where(s => s.Course.Teachers.Contains(userId)).OrderBy(m=>m.StartTime).ToListAsync();
+    }
```

Comment:
I think we can refactor this method into the one above, getting:
```suggestion
    public async Task<List<Session>> GetUpcomingSessions(string userId)
    {
        return await _context.Sessions
            .Include(s => s.Course)
            .Where(s => s.Course.Students.Contain(userId) || s.Course.Teachers.Contains(userId)).OrderBy(m=>m.StartTime).ToListAsync();
    }
```